navapbc/labs-decision-support-tool
# feat: Literal ai feedback

## Ticket

https://navalabs.atlassian.net/browse/DST-614


## Changes

> Added endpoint `feedback` to accept  `session_id`, `is_positive`, `response_id`, `comment` and `user_id`, send the values to literalai, and return a response value feedback created successfully

## Testing
1. run `make start`
2. query the api via localhost:8000/api/query using similar payload to the following:
```
{
  "session_id": "your session name",
  "new_session": "true",
  "message": "What is a base period for SDI?" 
}
```


3. make note of the `response_id`
4. call the feedback endpoint localhost:8000/api/feedback with the payload:

```
{
    "session_id": "your session name",
    "is_positive": "true",
    "response_id": response_id from previous call,
    "comment": "great"
}
```


5. results should be reflected in LiteralAI logs; if `is_positive` is "true", the score value should be 1, if it's "false" the score value should be 0
<img width="894" alt="Screenshot 2024-12-06 at 5 04 50 PM" src="https://github.com/user-attachments/assets/dc21c2c1-c136-4c02-8bb2-e18041b2bd3a">

<img width="1177" alt="Screenshot 2024-12-06 at 5 05 19 PM" src="https://github.com/user-attachments/assets/a8fa0b7b-b171-4d2b-b2dc-f246b40f7d21">
